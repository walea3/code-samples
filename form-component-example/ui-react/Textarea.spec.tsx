import { renderWithNEL } from '@nel-ui/tooling-react/test-utils';
import React from 'react';

import { Textarea } from '.';

describe('<Textarea />', () => {
  const requiredProps = {
    id: 'textarea-id',
  };

  describe('@renders', () => {
    test.each`
      name                                    | props
      ${'default'}                            | ${{}}
      ${'with className'}                     | ${{ className: 'test-textarea' }}
      ${'with label'}                         | ${{ label: 'Textarea label' }}
      ${'when disabled'}                      | ${{ disabled: true }}
      ${'when disabled + with warning alert'} | ${{ disabled: true, alert: 'warning' }}
      ${'when disabled + with error alert'}   | ${{ disabled: true, alert: 'error' }}
      ${'with warning alert'}                 | ${{ alert: 'warning' }}
      ${'with error alert'}                   | ${{ alert: 'error' }}
      ${'with data-ref'}                      | ${{ 'data-ref': 'uid' }}
      ${'with data-ref + label'}              | ${{ 'data-ref': 'uid', label: 'Textarea label' }}
      ${'with other props'}                   | ${{ 'data-custom': 'custom-value' }}
    `('$name', ({ props }) =>
      expect(renderWithNEL(<Textarea {...requiredProps} {...props} />)).toMatchSnapshot()
    );
  });
});
