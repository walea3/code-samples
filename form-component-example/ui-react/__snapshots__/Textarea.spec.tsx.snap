// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Textarea /> @renders default 1`] = `
<textarea
  aria-invalid="false"
  class="nel-textarea nel-input"
  data-ref="textarea"
  id="textarea-id"
/>
`;

exports[`<Textarea /> @renders when disabled + with error alert 1`] = `
<textarea
  aria-invalid="true"
  class="nel-textarea nel-input nel-disabledInput nel-errorInput"
  data-ref="textarea"
  disabled=""
  id="textarea-id"
/>
`;

exports[`<Textarea /> @renders when disabled + with warning alert 1`] = `
<textarea
  aria-invalid="true"
  class="nel-textarea nel-input nel-disabledInput nel-warningInput"
  data-ref="textarea"
  disabled=""
  id="textarea-id"
/>
`;

exports[`<Textarea /> @renders when disabled 1`] = `
<textarea
  aria-invalid="false"
  class="nel-textarea nel-input nel-disabledInput"
  data-ref="textarea"
  disabled=""
  id="textarea-id"
/>
`;

exports[`<Textarea /> @renders with className 1`] = `
<textarea
  aria-invalid="false"
  class="nel-textarea nel-input test-textarea"
  data-ref="textarea"
  id="textarea-id"
/>
`;

exports[`<Textarea /> @renders with data-ref + label 1`] = `
<div
  class="nel-fieldControlLabelContainer"
  data-ref="uid.fieldControlLabelContainer"
>
  <label
    class="nel-inputLabel"
    data-ref="uid.fieldControlLabel"
    for="textarea-id"
    id="textarea-idLabel"
  >
    Textarea label
  </label>
  <textarea
    aria-invalid="false"
    class="nel-textarea nel-input"
    data-ref="uid.textarea"
    id="textarea-id"
  />
</div>
`;

exports[`<Textarea /> @renders with data-ref 1`] = `
<textarea
  aria-invalid="false"
  class="nel-textarea nel-input"
  data-ref="uid.textarea"
  id="textarea-id"
/>
`;

exports[`<Textarea /> @renders with error alert 1`] = `
<textarea
  aria-invalid="true"
  class="nel-textarea nel-input nel-errorInput"
  data-ref="textarea"
  id="textarea-id"
/>
`;

exports[`<Textarea /> @renders with label 1`] = `
<div
  class="nel-fieldControlLabelContainer"
  data-ref="fieldControlLabelContainer"
>
  <label
    class="nel-inputLabel"
    data-ref="fieldControlLabel"
    for="textarea-id"
    id="textarea-idLabel"
  >
    Textarea label
  </label>
  <textarea
    aria-invalid="false"
    class="nel-textarea nel-input"
    data-ref="textarea"
    id="textarea-id"
  />
</div>
`;

exports[`<Textarea /> @renders with other props 1`] = `
<textarea
  aria-invalid="false"
  class="nel-textarea nel-input"
  data-custom="custom-value"
  data-ref="textarea"
  id="textarea-id"
/>
`;

exports[`<Textarea /> @renders with warning alert 1`] = `
<textarea
  aria-invalid="true"
  class="nel-textarea nel-input nel-warningInput"
  data-ref="textarea"
  id="textarea-id"
/>
`;
